src/components/ProductOrder/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import PhotoSwitcher from 'components/PhotoSwitcher';
+import ProductOrderInfo from 'components/ProductOrderInfo';
+import { useGetProductByIdQuery } from 'redux/productsApi';
+import SameStyleProducts from 'components/SameStyleProducts';
+import styles from './index.module.scss';
+
+const ProductOrder = () => {
+  const { productId } = useParams();
+  const { data } = useGetProductByIdQuery({ id: productId });
+
+  useEffect(() => {
+    try {
+      const visitedProduct = localStorage.getItem('visited');
+      const visitedProductArray = visitedProduct
+        ? JSON.parse(visitedProduct)
+        : [];
+      const newProduct = data?.id;
+
+      if (newProduct && !visitedProductArray.includes(newProduct)) {
+        visitedProductArray.unshift(newProduct);
+      }
+      localStorage.setItem('visited', JSON.stringify(visitedProductArray));
+    } catch (error) {
+      console.error('Error handling localStorage:', error);
+    }
+  }, [productId]);
```

Comment:
I have already added the useLocalStorage hook
```suggestion
useEffect(() => {
    const { getItem, setItem } = useLocalStorage<string[]>('visited', []);
    const { id } = data;
    const visitedProduct = getItem();

    if (id && !visitedProduct.includes(id)) {
      visitedProduct.unshift(id);
      setItem(visitedProduct);
    }
  }, [productId]);
```